agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -511,3 +519,17 @@ func (state *DockerTaskEngineState) TaskARNByV3EndpointID(v3EndpointID string) (
 	taskArn, ok := state.v3EndpointIDToTask[v3EndpointID]
 	return taskArn, ok
 }
+
+func (state *DockerTaskEngineState) GetECSClient() api.ECSClient {
+	state.lock.RLock()
+	defer state.lock.RUnlock()
+
+	return state.client
+}
+
+func (state *DockerTaskEngineState) SetECSClient(client api.ECSClient) {
+	state.lock.RLock()
```

Comment:
should be lock.Lock() instead of RLock().

Reply:
Fixed!